internal/indexcoord/index_coord.go

```diff
@@ -828,13 +828,13 @@ func (i *IndexCoord) watchMetaLoop() {
 			}
 			if err := resp.Err(); err != nil {
 				if err == v3rpc.ErrCompacted {
-					newMetaTable, internalErr := NewMetaTable(i.metaTable.client)
+					internalErr := i.metaTable.reloadFromKV()
 					if internalErr != nil {
 						log.Error("Constructing new meta table fails when etcd has a compaction error",
 							zap.String("path", indexFilePrefix), zap.String("etcd error", err.Error()), zap.Error(internalErr))
 						panic("failed to handle etcd request, exit..")
 					}
-					i.metaTable = newMetaTable
+					i.indexBuilder.reloadFromKV(i.nodeManager.ListAllNodes())
```

Comment:
looks like these two reload is not atomic, will there be any problem?

Reply:
indexBuilder will not load data from the etcd, it will only use metaTable. reloadFromKV creates a misunderstanding.